docs/cookbook/adding-a-new-elasticsearch-index.md

```diff
@@ -0,0 +1,294 @@
+declare(strict_types=1);
+
+namespace App\Model\Category\Elasticsearch;
+
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Shopsys\FrameworkBundle\Component\Elasticsearch\AbstractExportCronModule;
+use Shopsys\FrameworkBundle\Component\Elasticsearch\IndexDefinitionLoader;
+use Shopsys\FrameworkBundle\Component\Elasticsearch\IndexFacade;
+
+class CategoryExportCronModule extends AbstractExportCronModule
+{
+    /**
+     * @param \App\Model\Category\Elasticsearch\CategoryIndex $index
+     * @param \Shopsys\FrameworkBundle\Component\Elasticsearch\IndexFacade $indexFacade
+     * @param \Shopsys\FrameworkBundle\Component\Elasticsearch\IndexDefinitionLoader $indexDefinitionLoader
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     */
+    public function __construct(
+        CategoryIndex $index,
+        IndexFacade $indexFacade,
+        IndexDefinitionLoader $indexDefinitionLoader,
+        Domain $domain
+    ) {
+        parent::__construct($index, $indexFacade, $indexDefinitionLoader, $domain);
+    }
+}
+```
+
+Now if we have [crons](../introduction/cron.md) properly configured the new cron will be started automatically with others.
+Or you may want to [configure](../cookbook/working-with-multiple-cron-instances.md) this cron module with different timing
+
+### Implement partial update
+
+Sometimes we want to export data immediately after the original data are changed (hiding category, rename, ...).
+For this purpose we need to implement `CategoryIndex::getExportDataForIds()`, scheduler for queuing, and [subscriber](https://symfony.com/doc/current/event_dispatcher.html#creating-an-event-subscriber) for processing queue.
+
+Anywhere you want to add row to immediate export you may call `CategoryExportScheduler::scheduleRowIdForImmediateExport()` for adding a row into the queue.
+
+#### Scheduler
+
+Scheduler is used as a queue of IDs which we want to export. When we make any changes we may add an affected category ID into this queue and Subscriber will pick it up after request is done.
+
+Create class `CategoryExportScheduler` which extends `AbstractExportScheduler`. We do not need to override nor implement any method.
+
+```php
+declare(strict_types=1);
+
+namespace App\Model\Category\Elasticsearch;
+
+use Shopsys\FrameworkBundle\Component\Elasticsearch\AbstractExportScheduler;
+
+class CategoryExportScheduler extends AbstractExportScheduler
+{
+}
+```
+
+#### Subscriber
+
+Create class `CategoryExportSubscriber` which extends `AbstractExportSubsriber`.
+Override its `__construct()`, and `getSubscribedEvents()`.
+
+Here is important to override constructors arguments type hint.
+Instead of abstract classes from `\Shopsys\FrameworkBundle\Component\Elasticsearch` you need to replace it with our new implementation (`CategoryExportScheduler`, `CategoryIndex`).
+
+Implementation of `getSubscribedEvents()` is desired to use `exportScheduledRows()` from abstract class but also we can implement in by ourselves.
```

Comment:
The method  `getSubscribedEvents()` is abstract in the parent class. So it has to be implemented